src/fonduer/candidates/candidates.py

```diff
@@ -56,22 +59,62 @@ def apply(self, xs, split=0, **kwargs):
         """Call the CandidateExtractorUDF."""
         super(CandidateExtractor, self).apply(xs, split=split, **kwargs)
 
-    def clear(self, session, split, **kwargs):
+    def clear(self, split, **kwargs):
         """Delete Candidates of each class from given split the database."""
         for candidate_class in self.candidate_classes:
             logger.info(
                 "Clearing table {} (split {})".format(
                     candidate_class.__tablename__, split
                 )
             )
-            session.query(Candidate).filter(
+            self.session.query(Candidate).filter(
                 Candidate.type == candidate_class.__tablename__
             ).filter(Candidate.split == split).delete()
 
-    def clear_all(self, session, split, **kwargs):
+    def clear_all(self, split, **kwargs):
         """Delete all Candidates from given split the database."""
         logger.info("Clearing ALL Candidates.")
-        session.query(Candidate).filter(Candidate.split == split).delete()
+        self.session.query(Candidate).filter(Candidate.split == split).delete()
+
+    def get_candidates(self, docs=None, split=0):
```

Comment:
It seems this is highly redundant with get_mentions. Can we reuse most of the code?

Reply:
👍 good idea